purchase_ext_sst/models/purchase_order.py

```diff
@@ -133,13 +139,58 @@ def create(self, vals):
     @api.onchange('phone_search')
     def onchange_phone_search(self):
         if self.phone_search:
+            self.phone_update = False
+            self.mobile_update = False
             partner = self.get_partner_from_phone(self.phone_search)
             if partner:
                 self.partner_id = partner
-                self.phone_update = False
-                self.mobile_update = False
-            else:
-                self.phone_update = self.phone_search
+            elif self.partner_id:
+                if self.is_default_partner(self.partner_id.id):
+                    self.phone_update = self.phone_search
+
+    @api.onchange('phone_update')
+    def onchange_phone_update(self):
```

Comment:
I added this method to check whether the entered `phone_update` conflicts with existing records / current record (reset the field if it has conflicts)

Reply:
Good, but why don't we combine `onchange_phone_update()` and `onchange_mobile_update` together?

Reply:
I tried to combine the method, but it may cause some redundant execution:
e.g. when phone_update is filled and mobile_update is filled, there will be two search and the first one is actually redundant search.
```
@api.onchange('phone_update', 'mobile_update')
def onchange_phone_update(self):
	if self.phone_update:
		return self.check_onchange_phone(self.phone_update, 'phone_update')
	if self.mobile_update:
		return self.check_onchange_phone(self.mobile_update, 'mobile_update')
```

Reply:
Actually, doesn't it make more sense to have this check in `create_partner()` and `update_partner()`? It doesn't have to be done by onchange.

Reply:
Previously they were in `create()` and `write()`, ref: https://github.com/rfhk/sts-custom/pull/90/files/d8c75e892663eda5b17040afbb5d24c71c52fede#diff-c087fd7395c3d0535abd08c956939356R109

But I think it is more user-friendly to let the staff know about there is / are conflict(s) when they fill in the phone_update / mobile_update.